tests/end_to_end/netsh_test.cpp

```diff
@@ -534,3 +534,73 @@ TEST_CASE("unpin program", "[netsh][programs]")
     REQUIRE(result == NO_ERROR);
 }
+
+TEST_CASE("xdp interface parameter", "[netsh][programs]")
+{
+    _test_helper_libbpf test_helper;
+
+    // Load a program pinned.
+    int result;
+
+    // Load program with pinpath and loopback interface alias.
+    std::string output = run_netsh_command_with_args(
+        handle_ebpf_add_program, &result, 4, L"droppacket.o", L"xdp", L"mypinpath", L"Loopback Pseudo-Interface 1");
+    REQUIRE(strcmp(output.c_str(), "Loaded with ID 196609\n") == 0);
+    REQUIRE(result == NO_ERROR);
+    output = _run_netsh_command(handle_ebpf_delete_program, L"196609", nullptr, nullptr, &result);
+    REQUIRE(result == NO_ERROR);
+    REQUIRE(output == "Unpinned 196609 from mypinpath\n");
+    REQUIRE(bpf_object__next(nullptr) == nullptr);
+
+    // Load program with pinpath and loopback interface name.
+    output = run_netsh_command_with_args(
+        handle_ebpf_add_program, &result, 4, L"droppacket.o", L"xdp", L"mypinpath", L"loopback_0");
+    REQUIRE(strcmp(output.c_str(), "Loaded with ID 196610\n") == 0);
+    REQUIRE(result == NO_ERROR);
+    output = _run_netsh_command(handle_ebpf_delete_program, L"196610", nullptr, nullptr, &result);
+    REQUIRE(result == NO_ERROR);
+    REQUIRE(output == "Unpinned 196610 from mypinpath\n");
+    REQUIRE(bpf_object__next(nullptr) == nullptr);
+
+    // Load program loopback interface index.
+    output = _run_netsh_command(handle_ebpf_add_program, L"droppacket.o", L"xdp", L"interface=1", &result);
+    REQUIRE(strcmp(output.c_str(), "Loaded with ID 196611\n") == 0);
+    REQUIRE(result == NO_ERROR);
+    output = _run_netsh_command(handle_ebpf_delete_program, L"196611", nullptr, nullptr, &result);
+    REQUIRE(result == NO_ERROR);
+    REQUIRE(output == "Unpinned 196611 from DropPacket\n");
+    REQUIRE(bpf_object__next(nullptr) == nullptr);
+
+    // (Negative) Load program incorrect interface name.
+    output = _run_netsh_command(handle_ebpf_add_program, L"droppacket.o", L"xdp", L"interface=foo", &result);
+    REQUIRE(strcmp(output.c_str(), "Interface parameter is invalid.\n") == 0);
+    REQUIRE(result == ERROR_SUPPRESS_OUTPUT);
+    REQUIRE(bpf_object__next(nullptr) == nullptr);
+
+    // (Negative) Load program with program type that does not support interface parameter.
+    output = _run_netsh_command(handle_ebpf_add_program, L"bindmonitor.o", L"bind", L"interface=1", &result);
+    REQUIRE(
+        strcmp(
+            output.c_str(), "Interface parameter is not allowed for program types that dont support interfaces.\n") ==
+        0);
+    REQUIRE(result == ERROR_SUPPRESS_OUTPUT);
+    REQUIRE(bpf_object__next(nullptr) == nullptr);
+
+    // Add program with no interface parameter
+    output = _run_netsh_command(handle_ebpf_add_program, L"droppacket.o", nullptr, nullptr, &result);
+    REQUIRE(strcmp(output.c_str(), "Loaded with ID 196614\n") == 0);
+    REQUIRE(result == NO_ERROR);
+    // Detach the program.
```

Comment:
```suggestion

    // Detach the program.
```

Reply:
Won't fix. The 3 steps, load, detach, re-attach together created one test variation. So I dont want to add line break.